src/Messaging/MessageBuffer.php

```diff
@@ -157,7 +157,7 @@ public function frameCheck(FrameInterface $frame) {
                         return $this->newCloseFrame(Frame::CLOSE_BAD_PAYLOAD);
                     }
 
-                    return $this->newCloseFrame(Frame::CLOSE_NORMAL);
+                    return $this->newCloseFrame($closeCode);
```

Comment:
Do you want to return the frame you are checking here instead of creating a new one? If there is any information after the close code, the library user won't see it.